psamm/tests/test_randomsparse.py

```diff
@@ -0,0 +1,223 @@
+
+        self._strategy = randomsparse.GeneDeletionStrategy(
+            self._mm, self._assoc)
+
+    def test_method_get_all(self):
+        expected_total = {'gene_1', 'gene_2'}
+        self.assertEqual(set(self._strategy.get_total()), expected_total)
+
+    def test_method_tests_and_delete(self):
+        expected_genes = {'gene_1', 'gene_2'}
+        expected_reaciton_set = {'rxn_2'}
+        test_dict = {}
+        first_round = True
+        for gene, deleted_reactions in self._strategy.tests():
+            self._strategy.delete(gene, deleted_reactions)
+            if first_round:
+                self.assertEqual(deleted_reactions, set())
+                first_round = False
+            else:
+                self.assertEqual(deleted_reactions, {'rxn_2'})
+            test_dict[gene] = deleted_reactions
+
+        self.assertTrue(all(x in test_dict for x in expected_genes))
+        self.assertTrue(expected_reaciton_set in test_dict.values())
+
+
+class TestRandomSparse(unittest.TestCase):
+    def setUp(self):
+        self._database = DictDatabase()
+        self._database.set_reaction('rxn_1', parse_reaction('A[e] => B[e]'))
+        self._database.set_reaction('rxn_2', parse_reaction('B[e] => C[e]'))
+        self._database.set_reaction('rxn_3', parse_reaction('B[e] => D[e]'))
+        self._database.set_reaction('rxn_4', parse_reaction('C[e] => E[e]'))
+        self._database.set_reaction('rxn_5', parse_reaction('D[e] => E[e]'))
+        self._database.set_reaction('rxn_6', parse_reaction('E[e] =>'))
+        self._database.set_reaction('ex_A', parse_reaction('A[e] <=>'))
+
+        self._mm = MetabolicModel.load_model(
+            self._database, self._database.reactions)
+        self._assoc = {
+            'rxn_1': boolean.Expression('gene_1'),
+            'rxn_2': boolean.Expression('gene_2 or gene_3'),
+            'rxn_5': boolean.Expression('gene_3 and gene_4')
+        }
+
+        self._strategy_react = randomsparse.ReactionDeletionStrategy(self._mm)
+
+        self._strategy_gene = randomsparse.GeneDeletionStrategy(
+            self._mm, self._assoc)
+        self._obj_reaction = 'rxn_6'
+
+        try:
+            self._solver = generic.Solver()
+        except generic.RequirementsError:
+            self.skipTest('Unable to find an LP solver for tests')
+
+        self._prob = fluxanalysis.FluxBalanceProblem(self._mm, self._solver)
+
+    def test_random_sparse_reaction_strategy(self):
+        expected_essential = [
+            {'rxn_1', 'rxn_6', 'ex_A', 'rxn_2', 'rxn_4'},
+            {'rxn_1', 'rxn_6', 'ex_A', 'rxn_3', 'rxn_5'}
+        ]
+        expected_deleted = [{'rxn_3', 'rxn_5'}, {'rxn_2', 'rxn_4'}]
+
+        essential, deleted = randomsparse.random_sparse(
+            self._strategy_react,
+            self._prob,
+            self._obj_reaction,
+            flux_threshold=100)
+
+        self.assertTrue(essential in expected_essential)
+        self.assertTrue(deleted in expected_deleted)
+
```

Comment:
You can make a more strict test here, I think:

``` python
expected = {'rxn_3', 'rxn_5'}, {'rxn_2', 'rxn_4'}
essential, deleted = randomsparse.random_sparse(
    ...
self.assertTrue(expected == (essential, deleted) or expected == (deleted, essential))
```

This tests that the essential and deleted are in agreement and not containing duplicate elements.